### PR TITLE
Feature/give possibility to not download the dataset qualities

### PR DESCRIPTION
**Pull request by a-moadel**
#### What does this PR implement/fix? Explain your changes.
We can Ignore downloading the dataset qualities if its not exist while getting datasets.

#### How should this PR be tested?
Clear the cache , and calling get_dataset with download_qualities set to false : this result in qulaities.xml to not be downloaded.
Clear the cache , and calling get_dataset with download_qualities set to default : this result in qulaities.xml to be downloaded.

### CODE REVIEW

File: openml/datasets/functions.py [hunk 1]
```diff
@@ -290,6 +290,8 @@ def _name_to_id(
     error_if_multiple : bool (default=False)
         If `False`, if multiple datasets match, return the least recent active dataset.
         If `True`, if multiple datasets match, raise an error.
+    download_qualities : bool, optional
```
Comment (PGijsbers) [inline openml/datasets/functions.py:293]:
```suggestion
    download_qualities : bool, optional (default=True)
```

File: openml/datasets/functions.py [hunk 2]
```diff
@@ -326,6 +328,8 @@ def get_datasets(
         make the operation noticeably slower. Metadata is also still retrieved.
         If False, create the OpenMLDataset and only populate it with the metadata.
         The data may later be retrieved through the `OpenMLDataset.get_data` method.
+    download_qualities : bool, optional
```
Comment (PGijsbers) [inline openml/datasets/functions.py:331]:
```suggestion
    download_qualities : bool, optional (default=True)
```

**Review by PGijsbers — CHANGES_REQUESTED** [review]
@mfeurer Do you think the qualities should be loaded if they are present in cache? That would mean different results of the function depending on the cache state. I would think we can skip it altogether?
@a-moadel Please also add unit tests that test the feature works as intended (after we agree on the desired behavior).

I noticed the `pre-commit` workflow is not triggered. Does anyone have an idea why this would not trigger?

Closes #1009.

**Comment by mfeurer** [comment]
This looks good to me now. @PGijsbers you had a comment about persistent behavior, that's not yet addressed, right?

File: tests/test_datasets/test_dataset_functions.py [hunk 3]
```diff
@@ -433,6 +433,11 @@ def test__get_dataset_qualities(self):
         qualities_xml_path = os.path.join(self.workdir, "qualities.xml")
         self.assertTrue(os.path.exists(qualities_xml_path))
 
+    def test__get_dataset_qualities_skip_download(self):
+        qualities = _get_dataset_qualities_file(self.workdir, 2, False)
```
Comment (PGijsbers) [inline tests/test_datasets/test_dataset_functions.py:437]: Could you use keywords here (e.g. `download_qualities=False`)?

File: openml/datasets/functions.py [hunk 4]
```diff
@@ -405,7 +412,9 @@ def get_dataset(
         features_file = _get_dataset_features_file(did_cache_dir, dataset_id)
 
         try:
-            qualities_file = _get_dataset_qualities_file(did_cache_dir, dataset_id)
+            qualities_file = _get_dataset_qualities_file(
```
Comment (PGijsbers) [inline openml/datasets/functions.py:415]: I think it makes more sense to use `download_qualities` here and not call the `_get_dataset_qualities_file` if we don't want to process the qualities.

File: openml/datasets/functions.py [hunk 5]
```diff
@@ -290,6 +290,8 @@ def _name_to_id(
     error_if_multiple : bool (default=False)
         If `False`, if multiple datasets match, return the least recent active dataset.
         If `True`, if multiple datasets match, raise an error.
+    download_qualities : bool, optional (default=True)
+        If `True`, also download qualities.xml file. If false use the file if it was cached.
```
Comment (PGijsbers) [inline openml/datasets/functions.py:294]: Please update this and the other references where it says the cached version is used if `false` is passed, as this is not the behavior anymore.

**Review by PGijsbers — CHANGES_REQUESTED** [review]
First off, sorry for the delay. I think it works as intended now 🥳 however I do have a few minor comments that I would prefer to see addressed before merging. I'd like to see references to the old behavior updated, and I suggested a minor code change which puts the responsibility of not loading `qualities` in the `get_dataset` function instead. I don't think it makes sense to pass down a `bool` that signifies you just want `""` back - then we don't need to call down in the first place.

**Review by PGijsbers — APPROVED** [review]

**Pull request merged**